eclipse-jdt/eclipse.jdt.core
# feat: issue optional warning using '==' for wrapper types

## What it does
When comparing wrapper types or Strings using '==' this is done by reference instead of value. Usually this is in error (e.g. Long only caches values between -127 and +127). By flagging this as an optional warning or even error (configurable) these cases can be easily weeded out.

Fixes #2176

## How to test

Use offending code comparing two Long references or two Strings using '==', and set the optional problem to a level like information, warning or error. The compiler should flag the offending line with the message `Comparing primitive wrapper types or Strings using '=='`

## Author checklist
